src/data-workspace/inputs/date-input.test.js

```diff
@@ -0,0 +1,77 @@
+import { useConfig } from '@dhis2/app-runtime'
+import userEvent from '@testing-library/user-event'
+import React from 'react'
+import { useUserInfo } from '../../shared/index.js'
+import { render } from '../../test-utils/index.js'
+import { FinalFormWrapper } from '../final-form-wrapper.js'
+import { DateInput } from './date-input.js'
+
+jest.mock('../../shared/use-user-info/use-user-info.js', () => ({
+    useUserInfo: jest.fn(),
+}))
+
+jest.mock('@dhis2/app-runtime', () => {
+    const originalModule = jest.requireActual('@dhis2/app-runtime')
+
+    return {
+        ...originalModule,
+        useConfig: jest.fn(),
+    }
+})
+
+describe('date input field', () => {
+    const props = {
+        cocId: 'HllvX50cXC0',
+        deId: 'rkAZZFGFEQ7',
+        disabled: undefined,
+        fieldname: 'rkAZZFGFEQ7.HllvX50cXC0',
+        form: {},
+        locked: false,
+        onFocus: jest.fn(),
+        onKeyDown: jest.fn(),
+    }
+
+    beforeEach(() => {
+        useUserInfo.mockImplementation(() => ({
+            data: {
+                settings: {
+                    keyUiLocale: 'en',
+                },
+            },
+        }))
+    })
+
+    afterEach(jest.clearAllMocks)
+
+    it('renders date input component', async () => {
```

Comment:
Great with the test 🎉

It might be nice to have a test that checks that mutation (mocked) fires with correct date value when you select date / clear date. That might be a bit tricky to implement, but I guess that would also help with some of the logic for testing the other calendars. Could be good to add a separate ticket for adding tests if you think it's too much for the first implementation.

Reply:
Added a few test cases that show the mutate function being called with the correct date value on dateSelect and an empty string on clear. Doing this for gregorian, nepali and ethiopian calendars